cmd/countCG/countCG.go

```diff
@@ -0,0 +1,367 @@
+		endAlnPos = fasta.RefPosToAlnPosCounter(refSeq, region.ChromEnd, lastRefPos, lastAlnPos)
+
+		//add this information to bedAln
+		//newString := chrom + "\t" + strconv.Itoa(startAlnPos) + "\t" + strconv.Itoa(endAlnPos) + "\t" + name + "\n"
+		output = append(output, bed.Bed{Chrom: chrom, ChromStart: startAlnPos, ChromEnd: endAlnPos, Name: regName, FieldsInitialized: 4})
+
+		//reset the ref/alignment position pair to that of the current region, so that it can be used for the next region
+		lastRefPos = region.ChromEnd
+		lastAlnPos = endAlnPos
+	}
+	return output, nil
+}
+
+func countCpG(fastaFile string, alnBed []bed.Bed) (regionCpGs []bed.Bed, err error) {
+
+	chr := fasta.Read(fastaFile)
+
+	//check that there are not more than 1 fasta sequence
+	if len(chr) == 2 {
+		log.Fatalf("Error: expecting exactly one record in fasta file, but got 2. Are you trying to compare two sequences? If yes, use --compare mode.\n")
+	}
+
+	if len(chr) > 2 {
+		log.Fatalf("Error: expecting exactly one record in fasta file, but got %d. Using --compare mode will allow for comparison of 2 sequences, but not more than 2.\n", len(chr))
+	}
+
+	seq := chr[0].Seq
+	//if sequence is empty
+	if len(seq) == 0 {
+		log.Fatalf("Error: fasta sequence is empty.\n")
+	}
+
+	for _, region := range alnBed {
+		//get the following information
+		chrom := region.Chrom
+		start := region.ChromStart
+		end := region.ChromEnd
+		name := region.Name
+
+		//Find the current region in the fasta sequences:
+		//(Here, we assume that both the genome of interest and the ancestral genome are unmodified
+		//from the original multiple alignment, so the same alignment coordinates apply to both)
+		regSeq := seq[start:end]
+
+		//initialize number of each category of CpG sites to 0
+
+		cpgCount := 0
+
+		//for length of entire sequence defined by alignment start/end coordinates
+		for i := 0; i < len(regSeq)-1; i++ {
+			f1, f2 := regSeq[i], regSeq[i+1]
+			if f1 == dna.C && f2 == dna.G {
+				cpgCount++
+			}
+		}
+		regionCpGs = append(regionCpGs, bed.Bed{Chrom: chrom, ChromStart: start, ChromEnd: end, Name: name, Score: cpgCount, FieldsInitialized: 5})
+	}
+	return regionCpGs, nil
+}
+
+func compareCpGcount(fastaFile string, alnBed []bed.Bed) (regionCpGs []CpGcounts, err error) {
+	seqs := fasta.Read(fastaFile)
+
+	//define genome of interest and ancestral genome in the multiFa
+	firstSeq := seqs[0].Seq
+	secondSeq := seqs[1].Seq
+
+	//for each region in the alignment-converted bed file
+	for _, region := range alnBed {
+		//get the following information
+		chrom := region.Chrom
+		start := region.ChromStart
+		end := region.ChromEnd
+		name := region.Name
+
+		//Find the current region in the fasta sequences:
+		//(Here, we assume that both the genome of interest and the ancestral genome are unmodified
+		//from the original multiple alignment, so the same alignment coordinates apply to both)
+		firstRegseq := firstSeq[start:end]
```

Comment:
Do bed and golang slice indexing match up? I know start definitely match up. bed region.ChromEnd does not include the end base. Does firstSeq[first:end] include the end base?